### PR TITLE
chore(node): refactor pricing metrics

### PR DESCRIPTION
**Pull request by maqi**
## Description

reviewpad:summary

### CODE REVIEW

File: sn_networking/src/cmd.rs [hunk 1]
```diff
@@ -355,19 +355,13 @@ impl SwarmDriver {
...
             SwarmCmd::GetLocalStoreCost { key, sender } => {
                 cmd_string = "GetLocalStoreCost";
-                let record_exists = self
-                    .swarm
-                    .behaviour_mut()
-                    .kademlia
-                    .store_mut()
-                    .contains(&key);
-                let cost = if record_exists {
-                    NanoTokens::zero()
```
Comment (joshuef) [inline sn_networking/src/cmd.rs:346]: Have we not lost the 0 cost if record exists here?

**Comment by joshuef** [comment]
I see the main aim here is the `QuoteMetrics` and getting that verifiable. I think we should try that atop the separate changes to the pricing mechanics in #1497

File: sn_networking/src/cmd.rs (hunk 1 above)
Comment (maqi) [reply sn_networking/src/cmd.rs:346]: no, will still return Zero cost within the inner calculation.

**Comment by maqi** [comment]
> I see the main aim here is the `QuoteMetrics` and getting that verifiable. I think we should try that atop the separate changes to the pricing mechanics in #1497

Ok, will update this once PR 1497 merged.

**Review by joshuef — APPROVED** [review]

**Pull request merged**